logictest: better settings overrides

tbg opened the pull request:
We were using a manually updated cluster setting in logictests when
overriding settings, which was flaky. Even if it hadn't been flaky,
it's better to make logictests operate on as standard a cluster as
we can, so instead of overrides we now use "SET CLUSTER SETTING ..."
and wait until that has propagated everywhere.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17731)
<!-- Reviewable:end -->

tschottdorf commented:
@dt would you do the honor?

dt commented:
Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[pkg/sql/logictest/logic_test.go, line 718 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731#-KrlNrwj12kK3eZe3E_i:-KrlNrwj12kK3eZe3E_j:b-6lucb7) ([raw file](https://github.com/cockroachdb/cockroach/blob/b7094f86810bd48593f43ac001c5b836174fb710/pkg/sql/logictest/logic_test.go#L718)):*
> ```Go
> 		}
> 		// Wait until all servers are aware of the setting.
> 		for allOK := false; !allOK; allOK = true {
> ```

does this need a util.SucceedsSoon to avoiding hanging forever if we mess something up?

---

*[pkg/sql/logictest/testdata/logic_test/event_log, line 329 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731#-KrlN7DLsN2lBp5Yd_RR:-KrlN7DLsN2lBp5Yd_RS:b-gbihee) ([raw file](https://github.com/cockroachdb/cockroach/blob/b7094f86810bd48593f43ac001c5b836174fb710/pkg/sql/logictest/testdata/logic_test/event_log#L329)):*
> ```
> SELECT "targetID", "reportingID", "info"
> FROM system.eventlog
> WHERE "eventType" = 'set_cluster_setting' AND info NOT LIKE '%version%' AND info NOT LIKE '%sql.defaults.distsql%'
> ```

can we whitelist instead? this seems harder to maintain.

---

*[pkg/sql/logictest/testdata/logic_test/system, line 338 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731#-KrlNQ1sGeSm4aeOERMz:-KrlNQ1sGeSm4aeOERN-:b-qt9dnk) ([raw file](https://github.com/cockroachdb/cockroach/blob/2db0954cd180b54ba6c27d68e199798887de8078/pkg/sql/logictest/testdata/logic_test/system#L338)):*
> ```
> # This is somewhat surprising.
> query T
> select name from system.settings order by name
> ```

ditto here -- let's just whitelist the one value or values we want to see.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
TFTR!

---

Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions.

---

*[pkg/sql/logictest/logic_test.go, line 718 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731#-KrlNrwj12kK3eZe3E_i:-KrldmeZF4-2_eJNWDN8:bh7scxj) ([raw file](https://github.com/cockroachdb/cockroach/blob/b7094f86810bd48593f43ac001c5b836174fb710/pkg/sql/logictest/logic_test.go#L718)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

does this need a util.SucceedsSoon to avoiding hanging forever if we mess something up?
</blockquote></details>

If we mess up at this level, I think a test timeout is an appropriate way of catching it.

---

*[pkg/sql/logictest/testdata/logic_test/event_log, line 329 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731#-KrlN7DLsN2lBp5Yd_RR:-Krle-M1Z2YSqubdrL7x:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b7094f86810bd48593f43ac001c5b836174fb710/pkg/sql/logictest/testdata/logic_test/event_log#L329)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

can we whitelist instead? this seems harder to maintain.
</blockquote></details>

Done.

---

*[pkg/sql/logictest/testdata/logic_test/system, line 338 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731#-KrlNQ1sGeSm4aeOERMz:-KrleY_fFUR_jBp8zMkk:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/2db0954cd180b54ba6c27d68e199798887de8078/pkg/sql/logictest/testdata/logic_test/system#L338)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

ditto here -- let's just whitelist the one value or values we want to see.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731#-:-KrlfbtXqE6ut2dJxlbW:b-814g8k)*
<!-- Sent from Reviewable.io -->

dt commented:
Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[pkg/sql/logictest/logic_test.go, line 718 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731#-KrlNrwj12kK3eZe3E_i:-KrljebqkcpBgdZ9Rd3R:b-wik7f9) ([raw file](https://github.com/cockroachdb/cockroach/blob/b7094f86810bd48593f43ac001c5b836174fb710/pkg/sql/logictest/logic_test.go#L718)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

If we mess up at this level, I think a test timeout is an appropriate way of catching it.
</blockquote></details>

Sure, I'm all for catching it, I'd just rather catch it in 30s with a failure than after 15 minutes with a timeout.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/sql/logictest/logic_test.go, line 718 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731#-KrlNrwj12kK3eZe3E_i:-KrlnCxgGUmUX7PVlvcj:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b7094f86810bd48593f43ac001c5b836174fb710/pkg/sql/logictest/logic_test.go#L718)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

Sure, I'm all for catching it, I'd just rather catch it in 30s with a failure than after 15 minutes with a timeout.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731)*
<!-- Sent from Reviewable.io -->

dt commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 5 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17731#-:-KrlnPevUHms-SzEo5sS:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tbg committed:
logictest: better settings overrides

We were using a manually updated cluster setting in logictests when
overriding settings, which was flaky. Even if it hadn't been flaky,
it's better to make logictests operate on as standard a cluster as
we can, so instead of overrides we now use "SET CLUSTER SETTING ..."
and wait until that has propagated everywhere.
